Controller/Callback/Index.php

```diff
@@ -27,28 +27,36 @@ class Index implements \Magento\Framework\App\ActionInterface
      */
     private $request;
 
+    /**
+     * @var ResponseInterface
+     */
+    private $response;
+
     /**
      * @param Session $session
      * @param ProcessPayment $processPayment
      * @param RequestInterface $request
+     * @param ResponseInterface $response
      */
     public function __construct(
         Session $session,
         ProcessPayment $processPayment,
-        RequestInterface $request
+        RequestInterface $request,
+        ResponseInterface $response
     ) {
         $this->session = $session;
         $this->request = $request;
         $this->processPayment = $processPayment;
+        $this->response = $response;
     }
 
     /**
      * execute method
      */
-    public function execute()
+    public function execute(): ResponseInterface
     {
         $this->processPayment->process($this->request->getParams(), $this->session);
 
-        return;
+        return $this->response;
```

Comment:
Please use a result factory: \Magento\Framework\Controller\ResultFactory to create a unique instance of a result and return that result instead of a shared instance injected through constructor.

In this case the return type could be a raw.

Reply:
Done